HIS/his_checker.py

```diff
@@ -0,0 +1,138 @@
+    store_file = " >> complexity.txt"
+    os.system(complexity + str(file) + store_file)
+
+    filename = "complexity.txt"
+    infile = open(filename, 'r') 
+
+    lines = infile.readlines() 
+
+    for line in lines:
+        sline = line.strip()
+        sline = sline.split('\t')
+    
+        colonyCount = int(sline[0])
+        if colonyCount > cyclomatic_treshold:
+            print("Complexity exceeded (max." + str(cyclomatic_treshold) + "): "+ sline[0] + " at " + sline[5])
+  	
+    infile.close()
+    os.remove("complexity.txt")
+
+def process_comments(file):
+
+    tokei = "tokei -f "
+    store_file = " >> cloc.txt"
+    os.system(tokei + str(file) + store_file)
+
+    file_name = "cloc.txt"
+    infile = open(file_name, 'r') 
+
+    lines = infile.readlines() 
+    for line in lines:
+        sline = line.strip()
+        sline = sline.split(' ')
+        i=1
+        param = []
+        files = -1
+        lines = -1
+        code = -1
+        comments = -1
+        j = 0
+        if sline[0].startswith("/"):
+            while i<len(sline):
+                if (sline[i].isdigit()):
+                    if (files == -1): files = int(sline[i])
+                    elif (lines == -1): lines = int(sline[i])
+                    elif (code == -1): code = int(sline[i])
+                    elif (comments == -1): comments = int(sline[i])
+                    if (comments > -1 and code > 0 and (comments/code)<comment_density and sline[0].endswith(".c")): 
+                        print("Comment density smaller than " + str(comment_density) + ": " 
+                                + str(comments/code)+"  at "+sline[0])
+                    j = j+1
+                i = i+1
+  	
+    infile.close()
+    os.remove("cloc.txt")
+
+def process_goto(file):
+
+    qmcalc = "qmcalc "
+    store_file = " >> goto.txt"
+
+    os.system(qmcalc + str(file) + store_file)
+
+    file_name = "goto.txt"
+    infile = open(file_name, 'r') 
+
+    lines = infile.readlines() 
+    file_path = []
+
+    for line in lines:
+        sline = line.strip()
+        sline = sline.split('\t')
+        if sline[0].startswith('/'): 
+            file_path = sline
+        elif (int(sline[18]) > goto_treshold): 
+            os.system(str(file_path)+" "+str(sline[18]))    
+
+    infile.close()
+    os.remove("goto.txt")
+
+def process_calls(file):
+    
+    cflow = "cflow -l --depth=2 "
+    store_file = " >> cflow.txt"
+
+    if file.endswith(".c"):
+        os.system(cflow + str(file) + store_file)
+
+        filename = "cflow.txt"
+        infile = open(filename, 'r') 
+
+        lines = infile.readlines() 
+        file_path = []
+        call_counter = 0
+
+        for line in lines:
+            sline = line.strip()
+            sline = sline.split('}')
+            sline[0] = sline[0].lstrip("{")
+            sline[0] = sline[0].lstrip()
+            sline[1] = sline[1].lstrip()
+
+            if (int(sline[0]) == 0 and call_counter > 7):
```

Comment:
The metric threshold here is a magic number, but in other checks is defined globally?!

Reply:
Defined call_treshold as a global variable